doc/tutorials/enable_ivshmem.rst

```diff
@@ -1,229 +1,200 @@
+      :align: center
+      :class: drop-shadow
 
-.. note:: These commands are applicable to Linux-based guests with ``CONFIG_UIO`` and ``CONFIG_UIO_PCI_GENERIC`` enabled.
+#. Save the scenario and launch script.
 
-4. Finally, a user application can get the shared memory base address from
-   the ``ivshmem`` device BAR resource
-   (``/sys/class/uio/uioX/device/resource2``) and the shared memory size from
-   the ``ivshmem`` device config resource
-   (``/sys/class/uio/uioX/device/config``).
+#. Build ACRN, copy all the necessary files from the development computer to the
+   target system, and launch the Service VM and post-launched User VMs.
 
-   The ``X`` in ``uioX`` above, is a number that can be retrieved using the
-   ``ls`` command:
+#. To verify inter-VM shared memory communication:
 
-   - For VM1 use ``ls -lh /sys/bus/pci/devices/0000:00:06.0/uio``
-   - For VM2 use ``ls -lh /sys/bus/pci/devices/0000:00:05.0/uio``
+   a. As IVSHMEM is a PCI device that shares memory inside a virtual machine,
+      the application running on the VM can access it as a standard PCI device.
+      The application needs to design a data transfer notification mechanism
+      between the VMs. See :ref:`inter-vm_communication_ivshmem_app`. In this example, the application is ``ivshmem_test.bin``.
 
-HV-Land Example
-===============
+   #. Check the inter-VM shared memory vail. Run the following commands in
+      POST_STD_VM2:
 
-This example uses hv-land inter-VM communication between two
-Linux-based VMs (VM0 is a pre-launched VM and VM2 is a post-launched VM).
+      .. code-block:: console
 
-1. Make a copy of the predefined hybrid_rt scenario on whl-ipc-i5 (available at
-   ``acrn-hypervisor/misc/config_tools/data/whl-ipc-i5/hybrid_rt.xml``) and
-   configure shared memory for the communication between VM0 and VM2. The shared
-   memory name is ``hv:/shm_region_0``, and shared memory size is 2M bytes. The
-   resulting scenario XML should look like this:
+         root@acrn-Standard-PC-i440FX-PIIX-1996:~# modprobe uio
+         root@acrn-Standard-PC-i440FX-PIIX-1996:~# modprobe uio_pci_generic
+         root@acrn-Standard-PC-i440FX-PIIX-1996:~# echo "1af4 1110" >/sys/bus/pci/drivers/uio_pci_generic/new_id
+         root@acrn-Standard-PC-i440FX-PIIX-1996:~# ls /dev/uio0
+         /dev/uio0
 
-   .. code-block:: none
-      :emphasize-lines: 2,3
+         touch test.file && echo 132456 >> test.file #Create a file and save the file
 
-      <IVSHMEM>
-             <IVSHMEM_ENABLED>y</IVSHMEM_ENABLED>
-             <IVSHMEM_REGION>hv:/shm_region_0, 2, 0:2</IVSHMEM_REGION>
-      </IVSHMEM>
+   #. Run the following command in POST_STD_VM2 to send the file:
 
-2. Build ACRN based on the XML configuration for hybrid_rt scenario on whl-ipc-i5 board::
+      .. code-block:: console
 
-      make BOARD=whl-ipc-i5 SCENARIO=<path/to/edited/scenario.xml>
+         root@acrn-Standard-PC-i440FX-PIIX-1996:~# ./ivshmem_test.bin uio0 tx test.file
```

Comment:
this "ivhsmem_test.bin" executable doesn't exist and there are no instructions how to create it.